CliFx/CliApplication.cs

```diff
@@ -143,6 +143,38 @@ await applicationSchema
             return 0;
         }
 
+        /// <summary>
+        /// Handle <see cref="CommandException"/> differently from the rest because we want to
+        /// display it different based on the <see cref="CommandErrorDisplayOptions"/> associated
+        /// with the error.
+        /// </summary>
+        private int HandleCommandException(IReadOnlyList<string> commandLineArguments, CommandException commandException)
+        {
+            var commandErrorDisplayOptions = commandException.ErrorDisplayOptions;
+
+            if (commandErrorDisplayOptions.HasFlag(CommandErrorDisplayOptions.None))
+            {
+                return commandException.ExitCode;
```

Comment:
```suggestion
				// Return early because we don't want to print anything if
				// the display option is set to 'None'.
                return commandException.ExitCode;
```